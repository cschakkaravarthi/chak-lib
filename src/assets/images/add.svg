<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 10 10"><defs><style>.cls-1{fill:#757576;}.cls-2{fill:none;}</style></defs><title>add</title><g id="Layer_2" data-name="Layer 2"><g id="Layer_1-2" data-name="Layer 1"><g id="Group_256" data-name="Group 256"><g id="Group_253" data-name="Group 253"><g id="Path_12" data-name="Path 12"><rect class="cls-1" x="4.52" y="2.51" width="1" height="5"/></g><g id="Path_13" data-name="Path 13"><rect class="cls-1" x="2.52" y="4.51" width="5" height="1"/></g></g><g id="Rectangle_87" data-name="Rectangle 87"><rect class="cls-2" width="10" height="10"/><path class="cls-1" d="M10,10H0V0H10ZM1,9H9V1H1Z"/></g></g></g></g></svg>