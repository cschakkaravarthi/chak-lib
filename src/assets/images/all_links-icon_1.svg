<svg xmlns="http://www.w3.org/2000/svg" width="136.65" height="135.402" viewBox="0 0 136.65 135.402"><g transform="translate(19066.336 12580)"><path d="M121.487,84.2V54.882l9.443,9.492,1.985-1.974L121.487,50.912V28.536H97.743a28.261,28.261,0,0,0-56.457,0H17.543V60L8.1,50.5,6.113,52.477l11.429,11.49V84.2a17.743,17.743,0,1,0,14.9,29.179l19.368,6.912a17.73,17.73,0,0,0,35.405,0l19.369-6.911a17.741,17.741,0,1,0,14.9-29.182Zm13.553,17.684a14.812,14.812,0,0,1-1.365,6.2,11.028,11.028,0,0,0-8.108-6.579,6.149,6.149,0,0,0,.8-2.989,6.266,6.266,0,0,0-12.531,0,6.168,6.168,0,0,0,.794,2.983,11.03,11.03,0,0,0-8.133,6.585,14.948,14.948,0,1,1,28.546-6.2Zm-18.406-3.365a3.466,3.466,0,1,1,3.467,3.448A3.461,3.461,0,0,1,116.634,98.516ZM69.514,4.435A25.464,25.464,0,0,1,91.686,42.487,17.048,17.048,0,0,0,77.84,30.1a9.286,9.286,0,0,0,1.075-4.337,9.4,9.4,0,0,0-18.8,0A9.286,9.286,0,0,0,61.188,30.1,17.043,17.043,0,0,0,47.344,42.488,25.464,25.464,0,0,1,69.514,4.435Zm0,51a25.45,25.45,0,0,1-19.8-9.47c.048-6.977,5.959-12.7,13.465-13.3a9.363,9.363,0,0,0,12.678,0c7.506.6,13.418,6.323,13.467,13.3A25.45,25.45,0,0,1,69.514,55.435Zm-6.6-29.67a6.6,6.6,0,0,1,13.2,0,6.5,6.5,0,0,1-1.439,4.048c-.038,0-.077-.006-.115-.006v.144a6.526,6.526,0,0,1-10.094,0v-.143c-.038,0-.076,0-.113.006A6.486,6.486,0,0,1,62.913,25.765ZM18.943,86.926a14.925,14.925,0,0,1,13.591,21.157A11.031,11.031,0,0,0,24.4,101.5a6.164,6.164,0,0,0,.794-2.983,6.264,6.264,0,0,0-12.528,0,6.172,6.172,0,0,0,.8,2.989,11.026,11.026,0,0,0-8.108,6.578A14.925,14.925,0,0,1,18.943,86.926Zm-3.477,11.59a3.465,3.465,0,1,1,3.464,3.448A3.46,3.46,0,0,1,15.466,98.516Zm3.477,18.318a14.924,14.924,0,0,1-11.481-5.388c.137-4.1,3.852-7.391,8.411-7.391v-.119a6.058,6.058,0,0,0,6.138-.015v.134c4.559,0,8.276,3.3,8.411,7.391A14.919,14.919,0,0,1,18.943,116.834Zm1.4-32.636V63.965l11.429-11.49L29.787,50.5,20.342,60V31.336H41.286A28.29,28.29,0,0,0,68.115,58.164V101.6a17.771,17.771,0,0,0-16.242,15.738l-17.735-6.328A17.7,17.7,0,0,0,20.342,84.2Zm49.172,50.04a14.925,14.925,0,0,1-11.481-5.388c.134-4.1,3.851-7.394,8.411-7.394v-.116a6.058,6.058,0,0,0,6.138-.015v.132c4.561,0,8.277,3.3,8.414,7.392A14.931,14.931,0,0,1,69.514,134.238Zm-.014-14.87a3.449,3.449,0,1,1,3.464-3.448A3.46,3.46,0,0,1,69.5,119.368Zm13.606,6.116a11.031,11.031,0,0,0-8.136-6.583,6.152,6.152,0,0,0,.792-2.98,6.264,6.264,0,0,0-12.528,0,6.161,6.161,0,0,0,.8,2.986,11.026,11.026,0,0,0-8.109,6.579,14.954,14.954,0,1,1,27.184,0Zm4.05-8.143A17.771,17.771,0,0,0,70.914,101.6V58.164A28.29,28.29,0,0,0,97.743,31.336h20.945V50.911L107.257,62.4l1.985,1.974,9.446-9.495V84.2a17.7,17.7,0,0,0-13.8,26.814Zm32.932-.507a14.93,14.93,0,0,1-11.481-5.386c.137-4.1,3.852-7.392,8.411-7.392v-.134a6.058,6.058,0,0,0,6.138.015v.119c4.559,0,8.274,3.3,8.411,7.392A14.929,14.929,0,0,1,120.088,116.834Z" transform="translate(-19067.525 -12581.635)"/></g></svg>