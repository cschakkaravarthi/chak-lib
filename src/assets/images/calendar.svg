<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 24.0.1, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Group_522" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 19 20" style="enable-background:new 0 0 19 20;" xml:space="preserve">
<style type="text/css">
	.st0{fill:#2B2B2B;}
</style>
<g>
	<path class="st0" d="M17.5,2H16V0h-1v2H4V0H3v2H1.5C1.1,2,0.7,2.2,0.4,2.4C0.2,2.7,0,3.1,0,3.5V5v1v12.5C0,19.3,0.7,20,1.5,20h16
		c0.4,0,0.8-0.2,1.1-0.4c0.3-0.3,0.4-0.7,0.4-1.1v-15C19,2.7,18.3,2,17.5,2z M1,3.5c0-0.1,0-0.3,0.1-0.3C1.2,3.1,1.3,3,1.5,3h16
		C17.7,3,18,3.2,18,3.5V5H1V3.5z M17.9,18.8c-0.1,0.1-0.2,0.2-0.4,0.2h-16C1.2,19,1,18.8,1,18.5V6h17v12.5
		C18,18.6,18,18.7,17.9,18.8z"/>
	<path class="st0" d="M15,16h-3v1h3.5c0.3,0,0.5-0.2,0.5-0.5V13h-1V16z"/>
</g>
</svg>
