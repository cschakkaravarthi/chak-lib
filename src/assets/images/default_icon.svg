<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12.53 12.53"><defs><style>.cls-1{fill:#2b2b2b;fill-rule:evenodd;}</style></defs><title>Asset 1</title><g id="Layer_2" data-name="Layer 2"><g id="Layer_1-2" data-name="Layer 1"><path class="cls-1" d="M1.79,10.23,3,8.93,2.08,6.46,1.4,6a.13.13,0,0,1-.06-.14l.22-1.73A.15.15,0,0,1,1.67,4l2.7-.66.37-1.92a.15.15,0,0,1,.13-.12l.82-.09a.15.15,0,0,1,.16.13v0l0,.61.45-.23L6.38.31v0H6.26A6.25,6.25,0,0,0,5.1.4,5.82,5.82,0,0,0,4,.74H4A5.91,5.91,0,0,0,2,2H2A6,6,0,0,0,1.29,3,6.35,6.35,0,0,0,.74,4h0A5.82,5.82,0,0,0,.4,5.1,6.25,6.25,0,0,0,.29,6.26,6.36,6.36,0,0,0,.4,7.43,6,6,0,0,0,.74,8.55h0a6.25,6.25,0,0,0,.55,1,5.34,5.34,0,0,0,.5.65ZM6.67.3a6.31,6.31,0,0,1,.76.1A5.73,5.73,0,0,1,8.54.74h0a6.35,6.35,0,0,1,1,.55,6,6,0,0,1,.91.75h0l.15.16L8.83,2.11h0a.14.14,0,0,0-.12.16l.15.83L7.93,4h0a.23.23,0,0,0,0,.1l0,1.09L7,6.27s0,.05,0,.08a.15.15,0,0,0,.13.16l1.18.13a.14.14,0,0,0,.14-.07L8.73,6l.61.5h0a.15.15,0,0,0,.09,0h1v0l-1.72.52h0l-1.51.5h0a.14.14,0,0,0-.09.08L6.34,9.45h0a.15.15,0,0,0,.07.19l2.32,1v1.06l-.17.07h0a6,6,0,0,1-1.12.34,6.45,6.45,0,0,1-1.17.11,6.25,6.25,0,0,1-1.16-.11A5.82,5.82,0,0,1,4,11.79a6.35,6.35,0,0,1-1-.55A6,6,0,0,1,2,10.49L2,10.44,3.3,9.06a.16.16,0,0,0,0-.16l-1-2.59a.18.18,0,0,0-.05-.07h0l-.65-.42.2-1.54,2.69-.66a.14.14,0,0,0,.11-.11h0L5,1.6l.55-.06,0,.7a.15.15,0,0,0,.14.15.09.09,0,0,0,.07,0h0L6.46,2a.13.13,0,0,0,.08-.12h0L6.67.34v0ZM9,11.58c.19-.11.39-.22.57-.34a6,6,0,0,0,.91-.75h0a6,6,0,0,0,.75-.91,6.35,6.35,0,0,0,.55-1h0a5.73,5.73,0,0,0,.34-1.11,6.36,6.36,0,0,0,.11-1.17,6.25,6.25,0,0,0-.11-1.16A6.63,6.63,0,0,0,11.79,4a6,6,0,0,0-.55-1,5.3,5.3,0,0,0-.34-.45L9,2.4l.12.73h0a.11.11,0,0,1,0,.12l-.89.91,0,1.09a.16.16,0,0,1,0,.09l-.75.91.83.09.38-.6s0,0,0,0a.16.16,0,0,1,.21,0l.69.58h1a.15.15,0,0,1,.14.13l0,.28a.14.14,0,0,1-.1.16L8.77,7.4h0l-1.45.47L6.66,9.43l2.28,1h0a.16.16,0,0,1,.08.14v1ZM9.74,1.05a6.46,6.46,0,0,1,.95.78h0a5.48,5.48,0,0,1,.78,1,5.62,5.62,0,0,1,.58,1.08A5.87,5.87,0,0,1,12.41,5a6.31,6.31,0,0,1,0,2.45,6.48,6.48,0,0,1-.35,1.16h0a5.62,5.62,0,0,1-.58,1.08,5.88,5.88,0,0,1-.78.95h0a5.88,5.88,0,0,1-.95.78,6.56,6.56,0,0,1-.81.47h0l-.25.11h0a5.87,5.87,0,0,1-1.17.36,6.31,6.31,0,0,1-2.45,0,5.87,5.87,0,0,1-1.17-.36,5.62,5.62,0,0,1-1.08-.58,6,6,0,0,1-1-.78,6.46,6.46,0,0,1-.78-.95A6.42,6.42,0,0,1,.48,8.67h0A6.58,6.58,0,0,1,.12,7.49,6.31,6.31,0,0,1,.12,5,6.34,6.34,0,0,1,.47,3.87h0a7,7,0,0,1,.58-1.08,6.67,6.67,0,0,1,.78-1h0a6.55,6.55,0,0,1,1-.78A7,7,0,0,1,3.87.47h0A6.58,6.58,0,0,1,5,.12a6.31,6.31,0,0,1,2.45,0A6.48,6.48,0,0,1,8.65.47h0a7,7,0,0,1,1.08.58Zm2,7.49A.14.14,0,0,1,12,8.47Z"/></g></g></svg>