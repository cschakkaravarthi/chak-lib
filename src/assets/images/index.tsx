export const images = {
  bannerSrc: require('./banner.jpg'),
  logoSrc: require('./logo.png'),
  siteLogo: require('./umgc-logo.svg'),
  sectionIcon1: require('./all_links-icon_1.svg'),
  sectionIcon2: require('./all_links-icon_2.svg'),
  sectionIcon3: require('./all_links-icon_3.svg'),
  noNotification: require('./no_notif.svg'),
  addToContactsIcon: require('./add.svg'),
  faceBrowser: require('./face_browser.svg'),
  faceDocument: require('./face_document.svg'),
  faceHeart: require('./face_heart.svg'),
  faceLock: require('./face_lock.svg'),
  faceStop: require('./face_stop.svg'),
  chromeIcon: require('./Chrome-Logo-Icon.png'),
  safariIcon: require('./Safari_logo.png'),
  edgeIcon: require('./Microsoft-Edge.png'),
  remove: require('./remove.svg'),
  allLinksGrid: require('./all_links_grid.svg'),
  external: require('./external.svg'),
  gearIcon: require('./gear.svg'),
  unlikedIcon: require('./like.svg'),
  likedIcon: require('./liked.svg'),
  editIcon: require('./edit.svg'),
  defaultIcon: require('./default_icon.svg'),
  defaultIconCircle: require('./default_icon_circle.svg'),
  searchIconWhite: require('./search_white.svg'),
  apCard1: require('./ap-text.svg'),
  apCard2: require('./man-guitar.svg'),
  createCard: require('./people-hands.jpg'),
  cardsCouple: require('./cards-couple.png'),
  searchIconGrey: require('./search_grey.svg'),
  addIcon: require('./add.svg'),
  removeIcon: require('./remove.svg'),
  caretRight: require('./caret-right.svg'),
  caretLeft: require('./caret-left.svg'),
  chevronUp: require('./chevron-up.svg'),
  chevronDown: require('./chevron-down.svg'),
  bell: require('./bell.svg'),
  close: require('./close.svg'),
  calendar: require('./calendar.svg'),
  envelope: require('./envelope_grey.svg'),
  faceLockNew: require('./face_lock_new.svg')
};
