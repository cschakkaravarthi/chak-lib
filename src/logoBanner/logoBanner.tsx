import React, { FC } from 'react';
import Image from 'react-bootstrap/Image';
import Skeleton from 'react-loading-skeleton';

import '../../scss/_banner.scss';

export interface LogoBannerProps {
  logoSrc: string;
  bannerSrc: string;
  isFetching?: boolean;
}

export const LogoBanner: FC<LogoBannerProps> = props => {
  const { logoSrc, bannerSrc, isFetching = false } = props;

  return !isFetching ? (
    <div style={{ backgroundImage: `url(${bannerSrc})` }} className="banner-props">
      <div className="banner-props dark-linear-gradient d-flex justify-content-end">
        <div className="d-flex align-items-center mx-lg-5 mx-sm-5 mx-3">
          <Image className="logo-props" src={logoSrc} rounded />
        </div>
      </div>
    </div>
  ) : (
    <Skeleton height={145} />
  );
};
