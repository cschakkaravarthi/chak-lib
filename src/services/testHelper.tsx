/* eslint @typescript-eslint/no-explicit-any: "off" */
// In this one file we do actually want to use the "any" type. This is a very generic function.

import ShallowRenderer from 'react-test-renderer/shallow';

export const shallowRender = (toRender: any): Record<string, any> => {
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(toRender);
  return renderer.getRenderOutput();
};
