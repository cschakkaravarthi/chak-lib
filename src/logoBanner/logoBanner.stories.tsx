import React from 'react';
import StoriesContainer from '../StoriesContainer';
import { LogoBanner } from './logoBanner';
import { images } from '../assets/images';

export default {
  component: LogoBanner,
  title: 'Logo Banner'
};

export const logoBanner = (): JSX.Element => (
  <StoriesContainer>
    <LogoBanner bannerSrc={images.bannerSrc} logoSrc={images.logoSrc} />

    <hr />

    <LogoBanner bannerSrc={images.bannerSrc} logoSrc={images.logoSrc} isFetching={true} />
  </StoriesContainer>
);
