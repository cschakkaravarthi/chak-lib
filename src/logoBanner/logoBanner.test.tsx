import React from 'react';

import { shallowRender } from '../services/testHelper';
import { LogoBanner } from './logoBanner';
import { images } from '../assets/images';

describe('LogoBanner', () => {
  it('should render correctly', () => {
    const tree = shallowRender(
      <LogoBanner
        bannerSrc={images.bannerSrc}
        logoSrc={images.logoSrc}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
