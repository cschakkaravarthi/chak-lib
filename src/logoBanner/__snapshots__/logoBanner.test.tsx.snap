// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoBanner should render correctly 1`] = `
<div
  className="banner-props"
  style={
    Object {
      "backgroundImage": "url(test-file-stub)",
    }
  }
>
  <div
    className="banner-props dark-linear-gradient d-flex justify-content-end"
  >
    <div
      className="d-flex align-items-center mx-lg-5 mx-sm-5 mx-3"
    >
      <ForwardRef
        className="logo-props"
        fluid={false}
        rounded={true}
        roundedCircle={false}
        src="test-file-stub"
        thumbnail={false}
      />
    </div>
  </div>
</div>
`;
